crates/api/src/community/add_mod.rs

```diff
@@ -33,6 +33,18 @@ pub async fn add_mod_to_community(
     &mut context.pool(),
   )
   .await?;
+
+  // If its a mod removal, also check that you're a higher mod.
+  if !data.added {
+    check_is_higher_mod(
```

Comment:
wouldn't this prevent an admin from removing moderators if the admin isn't top mod?

i might be misreading this, but it seems that all the `check_is_higher_mod()` usages might be granting community mods immunity from some admin actions if the admin isn't top mod.

Reply:
That should be fixed by doing this:

```
let admins = local_user::table
  .filter(local_user::admin)
  .filter(local_user::person_id.eq_any(persons))
  .order_by(local_user::id)
  .select(local_user::person_id);

let moderators = community_moderator::table
  .filter(community_moderator::community_id.eq(for_community_id))
  .filter(community_moderator::person_id.eq_any(persons))
  .order_by(community_moderator::published)
  .select(community_moderator::person_id);

let res = admins
  .union_all(moderators)
  // This does a limit 1 select first
  .first::<PersonId>()
  .await?;
```

There should also be a unit test for this

Reply:
I have a `check_is_higher_mod_or_admin` as part of this, but it does both checks separately. I'll add this tho as its more performant.

Reply:
Also there are already unit tests for both these.